src/storage/core.rs

```diff
@@ -1048,18 +1066,33 @@ impl<K: Key + 'static> Safe<K> {
 }
 
 /// Trait `Key`
-pub trait Key: AsRef<[u8]> + Debug + Clone + Send + Sync + Ord + From<Vec<u8>> + Default {
+pub trait Key<'a>:
+    AsRef<[u8]> + From<Vec<u8>> + Debug + Clone + Send + Sync + Ord + Default
+{
     /// Key must have fixed length
     const LEN: u16;
 
+    /// Reference type for zero-copy key creation
+    type Ref: RefKey<'a>;
+
     /// Convert `Self` into `Vec<u8>`
     fn to_vec(&self) -> Vec<u8> {
         self.as_ref().to_vec()
     }
+
+    /// Convert `Self` to `Self::Ref`
+    fn as_ref_key(&'a self) -> Self::Ref {
+        Self::Ref::from(self.as_ref())
+    }
 }
 
+/// Trait for reference key type
+pub trait RefKey<'a>: Ord + From<&'a [u8]> {}
```

Comment:
Please add an empty line after the `RefKey` declaration to visually separate it from `BloomProvider`

Reply:
Done